apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java

```diff
@@ -262,6 +266,36 @@ private static void buildNPIResource(
     }
   }
 
+  /**
+   * Loads the taxonomy groupings into a map.
+   *
+   * @return map of taxonomy groupings.
+   * @throws IOException exception thrown.
+   */
+  public static Map<String, String> processTaxonomyDescriptions() throws IOException {
+    ClassLoader classLoader = DataUtilityCommons.class.getClassLoader();
+    File taxonomyFile =
+        new File(
+            Objects.requireNonNull(classLoader.getResource("nucc_taxonomy_240.csv")).getFile());
+    Map<String, String> taxonomyMapping = new HashMap<>();
+    try (FileInputStream is = new FileInputStream(taxonomyFile);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
```

Comment:
Would it be easier to use a CSV reader here?

Reply:
Maybe, but if we did that, then we should probably refactor the rest of the code in this module, as well, to use the CSV reader -- this follows the same pattern that already exists to read the NPI file. I'm not sure if introducing a library and refactoring all of this code is the best course of action, but I'm open to a conversation about it.

Reply:
Good call on the CSV reader. CSV parsing bugs were uncovered.

Reply:
Awesome, thanks for looking at that.

Reply:
Actually, I don't think this is always the case. I added this line
```java
 if (Integer.parseInt(entityTypeCode) == 2 && Strings.isNullOrEmpty(orgName)) {
            System.out.println("empty org");
 }
```
and it got a few hits before I stopped it.